zng/enum.go

```diff
@@ -0,0 +1,61 @@
+package zng
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/brimsec/zq/zcode"
+)
+
+type TypeEnum struct {
+	id       int
+	Type     Type
+	Elements []Element
+}
+
+type Element struct {
+	Name  string
+	Value zcode.Bytes
+}
+
+func NewTypeEnum(id int, typ Type, elements []Element) *TypeEnum {
```

Comment:
Would you please add a ztest that uses an enum? I didn't see one.

Reply:
Oops, good catch.  I found the enum test I thought I had committed and it reminded me I had another loose end to take care of... all done.

Reply:
BTW, there's some hokiness to the enum values in the type syntax...
```
#0:record[e:enum[int32,foo:[1],bar:[2],baz:[4]]]
```
This is the first time value syntax appeared in the type syntax.
I wrapped the values here in [] since this is escaped by the tzng value syntax and commas aren't. It's a bit ugly.  I had one thought here: we could change the tzng value encoding to use comma separation instead of semi-colon termination (and otherwise keep brackets) so that the type syntax and value syntax is consistent (@philrz suggested this a year ago).  Now would be the time to change this since it will break tzng format (it also messes up the azng updater).  The good news is only two small places in the code would need to change (but many tests would need to be updated).